### DISCUSSION TITLE
[save-sheets] investigate how other tools distinguish file paths and directory paths

### DISCUSSION THREAD

**Discussion by anjakefala** [saulpw/visidata#1191] [Ideas]
Saving multiple sheets in non-embedded formats requires the passing of a directory path. VisiData will create the directory, and save the sheets within it.

Currently, VisiData requires that the directory path passed through input ends in `/`. It will error otherwise.

## What do other tools do? What would you prefer VisiData to do?

What do other tools do? My guesses are:
* other tools require the directory to exist, and can then use `Path.is_dir()`
* other tools are explicitly expecting a directory or a file, and not trying to guess whether a path is a file or a directory
* they are also requiring `/` at the end

VisiData specific alternative suggestion
* maybe from the sheet format, and save_filetype whether a directory or a file is going to be required
* If a savefunc requires a directory path for more than one vsheets, or it will fail, one option is to assume the user is passing the name of a directory, and go with that.

**Comment by geekscrapy** [comment]
A sub question to this that I've not tried.

How would you do a `save-sheets`, saving to xlsx, but you _want_ to save to multiple files?

**Reply by saulpw** [reply]
That would be a separate command, something like
```
path = input(...)
for x in vd.stackedSheets:
    vd.saveSheets(path/vs.name, vs)
```

Maybe `save-each`?

**Reply by geekscrapy** [reply]
Yup, sounds good

**Comment by geekscrapy** [comment]
In terms of GUI apps with a save dialogue, if they are expecting a directory they won't allow a selection of files.

Another terminal app I use internally expects a directory path and even if you give it file.csv, it will create a dir of that name. This seems fine and it works well, and you realise soon enough
